### PR TITLE
CSCFAIRMETA-684: [ADD] Integrate Etsin & Qvain with Fairdata SSO

### PR DESCRIPTION
**Pull request by MattiasLevlin**
- Integrate Etsin login with Fairdata SSO

### CODE REVIEW

**Comment by MattiasLevlin** [comment]
To do:
- Set all required SSO session variables in authentication.py
- Account for different environments (test/stable/demo...) in variable names/functions

**Comment by MattiasLevlin** [comment]
Correction to previous commit: - There are ida_projects and idm_groups, but no **ida_groups**

**Comment by MattiasLevlin** [comment]
Functional now:
- Login/logout
- Retrieving datasets
- Publishing datasets

To do:
- Enable adding files from IDA
- Account for different environments (test/stable/demo...) in variable names/functions

Related [WIP] PR: https://github.com/CSCfi/etsin-ops/pull/31

**Comment by MattiasLevlin** [comment]
Prerequisites:
https://github.com/CSCfi/etsin-ops/pull/32

For testing instructions, see:
https://wiki.eduuni.fi/display/cscetsin/Etsin%3A+SSO

Description:
- Integrates Etsin and Qvain with Fairdata SSO
- For now, both use Etsin authentication

What should be verified as functional:
- Login/logout
- Publishing and updates to datasets
- Adding files from IDA using v1 & v2 filepickers
- Deleting files and datasets

**Review by tahme — CHANGES_REQUESTED** [review]
Looks good. Tested locally that login works, and when Etsin and Qvain Light run in separate `.local.fd-test.csc.fi` domains, logging into Etsin logs you also into Qvain as it's supposed to work. I have just a few minor cleanup suggestions.

I did notice that the relay parameter of the Etsin /sso endpoint is not working with SSO. It's the parameter that redirects you back to the page where you were after you login. That can be fixed in a separate issue.

File: etsin_finder/frontend/js/components/dataset/events/index.jsx [hunk 1]
```diff
@@ -167,7 +167,7 @@ class Events extends Component {
     let url = ''
 
     if (process.env.NODE_ENV === 'test') { /* test and stable */
-      url = `https://etsin-test.fairdata.fi/dataset/${identifier}`
+      url = `https://etsin.fd-test.csc.fi/dataset/${identifier}`
```
[the remark this answers is not in the document]
Comment (MattiasLevlin) [reply etsin_finder/frontend/js/components/dataset/events/index.jsx]: Addressed in 7845232

File: etsin_finder/qvain_light_utils.py [hunk 2]
```diff
@@ -497,15 +478,15 @@ def check_if_data_in_user_IDA_project(data):
...
+    user_ida_projects = get_user_ida_projects()
 
     # If user_ida_projects do not exist, there cannot be any data permission violations, so return True in this case
     if user_ida_projects is None:
         return True
 
     # Check IDA project permissions, if user_ida_projects exist
     try:
-        user_ida_projects_ids = [project.split(":")[1] for project in user_ida_projects]
+        user_ida_projects_ids = [project for project in user_ida_projects]
```
[the remark this answers is not in the document]
Comment (MattiasLevlin) [reply etsin_finder/qvain_light_utils.py]: Addressed in 84a398b

**Comment by MattiasLevlin** [comment]
> I did notice that the relay parameter of the Etsin /sso endpoint is not working with SSO. It's the parameter that redirects you back to the page where you were after you login. That can be fixed in a separate issue.

Good catch. This can be addressed in the next PR (CSCFAIRMETA-685), which should separate the Qvain login from the Etsin one.

**Comment by MattiasLevlin** [comment]
Commit 644c7ad requires https://github.com/CSCfi/etsin-ops/pull/33/files and addresses issues with logging in in test and demo, if the direct proxy is used

File: etsin_finder/frontend/js/components/general/loggedInUser.jsx [hunk 3]
```diff
@@ -7,14 +7,16 @@ import { faUser } from '@fortawesome/free-solid-svg-icons'
 
 const LoggedInUser = ({ Stores }) => {
   const { user, userLogged } = Stores.Auth
-  const { firstName } = user
+  const { firstName, lastName } = user
```
Comment (tahme) [inline etsin_finder/frontend/js/components/general/loggedInUser.jsx:10]: I think we should only show the first name. We initially tried to show both names but it took too much space. Or if it's possible that only one of them is defined, maybe show `firstName || lastName`.

**Review by tahme — CHANGES_REQUESTED** [review]

**Review by tahme — APPROVED** [review]
:+1:

**Pull request merged**